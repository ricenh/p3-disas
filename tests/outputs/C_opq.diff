--- outputs/C_opq.txt	2023-11-02 16:25:05.580163713 -0400
+++ expected/C_opq.txt	2023-10-10 11:54:35.000000000 -0400
@@ -1,11 +1,8 @@
-Usage: ../y86 <option(s)> mini-elf-file
- Options are:
-  -h      Display usage
-  -H      Show the Mini-ELF header
-  -a      Show all with brief memory
-  -f      Show all with full memory
-  -s      Show the program headers
-  -m      Show the memory contents (brief)
-  -M      Show the memory contents (full)
-  -d      Disassemble code contents
-  -D      Disassemble data contents
+Disassembly of executable contents:
+  0x100:                               | .pos 0x100 code
+  0x100:                               | _start:
+  0x100: 60 23                         |   addq %rdx, %rbx
+  0x102: 61 01                         |   subq %rax, %rcx
+  0x104: 62 45                         |   andq %rsp, %rbp
+  0x106: 63 67                         |   xorq %rsi, %rdi
+
