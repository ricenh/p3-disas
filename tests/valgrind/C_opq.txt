==2834541== Memcheck, a memory error detector
==2834541== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==2834541== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==2834541== Command: ../y86 -d inputs/opq.o
==2834541== 
Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 60 23                         |   addq %rdx, %rbx
  0x102: 61 01                         |   subq %rax, %rcx
  0x104: 62 45                         |   andq %rsp, %rbp
  0x106: 63 67                         |   xorq %rsi, %rdi

==2834541== 
==2834541== HEAP SUMMARY:
==2834541==     in use at exit: 0 bytes in 0 blocks
==2834541==   total heap usage: 4 allocs, 4 frees, 20,952 bytes allocated
==2834541== 
==2834541== All heap blocks were freed -- no leaks are possible
==2834541== 
==2834541== For lists of detected and suppressed errors, rerun with: -s
==2834541== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
