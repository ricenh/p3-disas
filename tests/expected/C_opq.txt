Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 60 23                         |   addq %rdx, %rbx
  0x102: 61 01                         |   subq %rax, %rcx
  0x104: 62 45                         |   andq %rsp, %rbp
  0x106: 63 67                         |   xorq %rsi, %rdi

